### PR TITLE
Bjorncs/container thread pool

### PR DESCRIPTION
**Pull request by bjorncs**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: container-core/src/main/java/com/yahoo/container/handler/threadpool/ContainerThreadPool.java [hunk 1]
```diff
@@ -83,7 +83,9 @@ private static int computeMaximumThreadPoolSize(int maxNumThreads) {
     }
 
     private static int computeCoreThreadPoolSize(int corePoolSize, int maxNumThreads) {
-        return Math.min(corePoolSize, maxNumThreads);
+        return corePoolSize <= 0
+            ? Runtime.getRuntime().availableProcessors() * 2
+            : Math.min(corePoolSize, maxNumThreads);
```
Comment (baldersheim) [inline container-core/src/main/java/com/yahoo/container/handler/threadpool/ContainerThreadPool.java:88]: I think that you need to do Math.min after you computed corePoolSize again.

Like
{code}
if (corePoolSize <= 0) corePoolSize = Runtime.getRuntime().availableProcessors() * 2 : corePoolSize;
Math.min(corePoolSize, maxNumThreads);
{code}

**Comment by baldersheim** [comment]
I also suggest that you run a system test or 2 first to verify.

**Comment by bjorncs** [comment]
Rebasing with master to resolve merge conflict with `ModelContextImpl`

**Comment by bjorncs** [comment]
@baldersheim PTAL. The two last commits are new since previous review.

**Review by baldersheim — APPROVED** [review]
Merge at will

**Comment by bjorncs** [comment]
Another merge conflict...

**Comment by bjorncs** [comment]
Rebased with master - no other changes.

**Pull request merged**